karyon3-archaius2/src/main/java/com/netflix/karyon/archaius/ArchaiusProxyAutoBinder.java

```diff
@@ -3,28 +3,39 @@
 import javax.inject.Inject;
 import javax.inject.Provider;
 
-import com.google.inject.Binder;
+import com.google.inject.AbstractModule;
 import com.google.inject.Key;
+import com.google.inject.Module;
 import com.netflix.archaius.ConfigProxyFactory;
-import com.netflix.karyon.spi.AutoBinder;
+import com.netflix.archaius.annotations.Configuration;
+import com.netflix.karyon.KeyMatchers;
+import com.netflix.karyon.spi.AbstractAutoBinder;
 
 /**
  * Autobinder to create proxies for any interface containing the archaius
  * annotations
  */
-final class ArchaiusProxyAutoBinder implements AutoBinder {
+final public class ArchaiusProxyAutoBinder extends AbstractAutoBinder {
+    public ArchaiusProxyAutoBinder() {
+        super(KeyMatchers.annotatedWith(Configuration.class));
+    }
+
     @Override
-    public <T> boolean configure(Binder binder, Key<T> key) {
-        binder.bind(key).toProvider(new Provider<T>() {
-            @Inject
-            ConfigProxyFactory factory;
-            
-            @SuppressWarnings("unchecked")
+    public <T> Module getModuleForKey(final Key<T> key) {
+        return new AbstractModule() {
             @Override
-            public T get() {
-                return (T) factory.newProxy(key.getTypeLiteral().getRawType());
+            protected void configure() {
+                bind(key).toProvider(new Provider<T>() {
+                    @Inject
+                    ConfigProxyFactory factory;
```

Comment:
Is this easily tested?  It seems if the outer class injected the ConfigProxyFactory through the constructor and allowed this inner class to reference it that you'd then be able to mock it as you'd like in tests without needing to set up a full injector context.